compiler/AST/UseStmt.cpp

```diff
@@ -175,6 +159,41 @@ bool UseStmt::hasExceptList() const {
 *                                                                             *
 ************************************** | *************************************/
 
+// Deprecated by Vass in 1.31: given `use BoundedRangeType`,
+// redirect it to `use boundKind`, with a deprecation warning.
+// This seems to handle `import BoundedRangeType` as well.
+static void checkRangeDeprecations(ResolveScope* scope, UseStmt* use,
```

Comment:
It's kind of unfortunate that we have to add this checking code to the compiler.

In the range module you added this:

```chpl
proc BoundedRangeType type do return boundKind
```

But I guess that doesn't fix things for a statement like this:

```chpl
use BoundedRangeType;
```

because we don't actually evaluate `BoundedRangeType` at compile-time before evaluating the `use` statement (so it thinks you're trying to pass a proc to the `use` statement). Could we / should we change this behavior? Having to modify compiler code for this case isn't just a pain for us, it also means there's no way for our users to deprecate and rename enums in their own modules.

That said, I don't think I don't think I'd try to generalize things for this PR, but if there isn't already a Github issue on this (e.g. "how can we rename an enum and its members and keep the old one around deprecated without having to modify compiler code") then I'd suggest creating one and having that cite back to this code review.

Reply:
Good suggestion. I created #22069, with pointers to two prior related issues.

Reply:
I don't think the compiler change is necessary - can't you just make the old name a type alias to the new enum and deprecate the type alias?

Reply:
If you can, it would be great to add a test of this to `test/deprecated-keyword`, either extending one of the enum examples or adding a new one.  It might also be helpful to add it as a word-around example to https://chapel-lang.org/docs/latest/developer/bestPractices/Deprecation.html#best-practices-deprecation in the meanwhile

Reply:
> can't you just make the old name a type alias to the new enum and deprecate the type alias?

Nice suggestion. Alas this is not supported today:
```
error: use must name a module or enum ('BoundedRangeType' is neither)
```

> it would be great to add a test of this to test/deprecated-keyword

I consider this throw-away code and do not see a justification to spend any effort on maintaining and testing it.

>  It might also be helpful to add it as a work-around example

Sure, please review: #22077